app/models/course/assessment.rb

```diff
@@ -96,11 +97,16 @@ def assign_folder_attributes
   end
 
   def set_defaults
-    self.draft ||= true
+    self.draft = true unless defined?(draft)
```

Comment:
I think `draft` is always defined ?  let's standard this and do not use `||=` to initialise booleans:
`self.draft = true if draft.nil?`